databuilder/extractor/dashboard/tableau/tableau_dashboard_utils.py

```diff
@@ -0,0 +1,209 @@
+                      re.sub(r'\.', '_',
+                             re.sub(r'(\[|\]|\(|\)|\-|\&|\?)', '', str)))
+
+    @staticmethod
+    def sanitize_database_name(str):
+        # type: (str) -> str
+        """
+        Sanitizes a given string so that it can safely be used as a table's database.
+        Sanitization behaves as follows:
+            - all hyphens are deleted
+        """
+        return re.sub(r'-', '', str)
+
+    @staticmethod
+    def sanitize_table_name(str):
+        """
+        Sanitizes a given string so that it can safely be used as a table name.
+        Replicates the current behavior of sanitize_workbook_name, but this is purely coincidental.
+        As more breaking characters/patterns are found, each method should be updated to reflect the specifics.
+        Sanitization behaves as follows:
+            - all forward slashes and single quotes characters are deleted
+        """
+        # type: (str) -> str
+        return re.sub(r'(\/|\')', '', str)
+
+    @staticmethod
+    def sanitize_workbook_name(str):
+        # type: (str) -> str
+        """
+        Sanitizes a given string so that it can safely be used as a workbook ID.
+        Mimics the current behavior of sanitize_table_name for now, but is purely coincidental.
+        As more breaking characters/patterns are found, each method should be updated to reflect the specifics.
+        Sanitization behaves as follows:
+            - all forward slashes and single quotes characters are deleted
+        """
+        return re.sub(r'(\/|\')', '', str)
+
+
+class TableauGraphQLApiExtractor(Extractor):
+    """
+    Base class for querying the Tableau Metdata API, which uses a GraphQL schema.
+    """
+
+    API_VERSION = const.API_VERSION
+    TABLEAU_HOST = const.TABLEAU_HOST
+    SITE_NAME = const.SITE_NAME
+    TABLEAU_ACCESS_TOKEN_NAME = const.TABLEAU_ACCESS_TOKEN_NAME
+    TABLEAU_ACCESS_TOKEN_SECRET = const.TABLEAU_ACCESS_TOKEN_SECRET
+    EXCLUDED_PROJECTS = const.EXCLUDED_PROJECTS
+    EXTERNAL_CLUSTER_NAME = const.EXTERNAL_CLUSTER_NAME
+    EXTERNAL_SCHEMA_NAME = const.EXTERNAL_SCHEMA_NAME
+    EXTERNAL_TABLE_TYPES = const.EXTERNAL_TABLE_TYPES
+    CLUSTER = const.CLUSTER
+    DATABASE = const.DATABASE
+    QUERY = 'query'
+    QUERY_VARIABLES = 'query_variables'
+
+    def init(self, conf):
+        self._conf = conf
+        self._auth_token = self._conf.get(TableauDashboardAuth.AUTH).token
+        self._query = self._conf.get(TableauGraphQLApiExtractor.QUERY)
+        self._iterator = None
+        self._static_dict = conf.get(STATIC_RECORD_DICT, dict())
+        self._metadata_url = 'https://{TABLEAU_HOST}/api/metadata/graphql'.format(
+            TABLEAU_HOST=self._conf.get_string(TableauGraphQLApiExtractor.TABLEAU_HOST)
+        )
+
+        try:
+            self._query_variables = self._conf.get(TableauGraphQLApiExtractor.QUERY_VARIABLES)
+        except ConfigMissingException:
+            self._query_variables = {}
```

Comment:
I think you can just
```suggestion
        self._query_variables = self._conf.get(TableauGraphQLApiExtractor.QUERY_VARIABLES, {})
```

Reply:
This is much better, didn't know that `.get()` took a default. TIL! +1